saleor/csv/utils/export.py

```diff
@@ -84,13 +92,68 @@ def get_product_queryset(scope: Dict[str, Union[str, dict]]) -> "QuerySet":
     elif "filter" in scope:
         queryset = ProductFilter(data=scope["filter"], queryset=queryset).qs
 
-    queryset = queryset.order_by("pk").prefetch_related(
-        "attributes", "variants", "collections", "images", "product_type", "category"
-    )
+    queryset = queryset.order_by("pk")
 
     return queryset
 
 
+def export_products_in_batches(
+    queryset: "QuerySet",
+    export_info: Dict[str, list],
+    export_fields: Set[str],
+    headers: List[str],
+    csv_headers_mapping: Dict[str, str],
+    delimiter: str,
+    export_file: ExportFile,
+    file_name: str,
+    file_type: str,
+):
+    products_count = queryset.count()
+    warehouses = export_info.get("warehouses")
+    attributes = export_info.get("attributes")
+
+    create_file = True
+    counter = 0
+    while counter < products_count:
+        qs = queryset[counter : counter + BATCH_SIZE]
+        # filter cancel limiting the query, so we need to ensure that only part of
+        # products will be fetch from database
+        product_batch = (
+            Product.objects.filter(pk__in=qs.values_list("pk", flat=True))
+            .order_by("pk")
+            .prefetch_related(
+                "attributes",
+                "variants",
+                "collections",
+                "images",
+                "product_type",
+                "category",
+            )
+        )
+
+        export_data = get_products_data(
+            product_batch, export_fields, warehouses, attributes,
+        )
+
+        if create_file:
+            create_csv_file_and_save_in_export_file(
```

Comment:
Question: What should we do in a case when `products_count` == 0? What will happen with `send_email_with_link_to_download_csv` when file will be empty?
- Do we want to create an empty file only with headers? 
- we don't want to create a file in that case?

Suggestion: If you would have a function for `create_csv_file_and_headers`, then you could call it before loop and this `if...else` condition would not be needed.

Reply:
IMO we should create the empty file only with headers. I'll test it for such a case.

Reply:
We can also check what other systems do in such a case. I expect that when having 0 products the export operation will be blocked at least in UI.

Reply:
But I think we still should handle this situation because right now it fails if there are no products.

Reply:
Yes, API should handle that as well. I guess creating an empty file is acceptable but I would check what other systems do. As I said - it should be blocked in UI so it's not common case that someone runs the export though API only.